### PR TITLE
kernel/modules-closure.sh v2: fix the modules closure

### PR DESCRIPTION
**Pull request by xaverdh**
Prior to this commit the modules closure could reference nonsensical
pieces of data in to form of module parameters. These were treated
as seperate modules.

"modprobe --show-depends" outputs a line of the form
"insmod module_name module_args", we are only interested in the
module name, so we extract it with sed.

Also fixes a bug in the original implementation where the check for missing modules did not work as indended (the exit status of
modprobe was ignored due to a pipe).

### CODE REVIEW

File: pkgs/build-support/kernel/modules-closure.sh [hunk 1]
```diff
@@ -22,10 +22,13 @@ echo "kernel version is $version"
...
 for module in $rootModules; do
     echo "root module: $module"
-    deps=$(modprobe --config no-config -d $kernel --set-version "$version" --show-depends "$module" \
-        | sed 's/^insmod //') \
-        || if test -z "$allowMissing"; then exit 1; fi
+    # Obtain the module dependencies with modprobe.
+    rawdeps=$(modprobe --config no-config -d "$kernel" --set-version "$version" --show-depends "$module") \
+     || if test -z "$allowMissing"; then exit 1; fi
+
     if [[ "$deps" != builtin* ]]; then
```
Comment (Infinisil) [inline pkgs/build-support/kernel/modules-closure.sh]: `$deps` seems to be used before it being declared now

Comment (xaverdh) [reply pkgs/build-support/kernel/modules-closure.sh]: Indeed, should be `rawdeps` now.

**Comment by xaverdh** [comment]
I kind of forgot about this one, although the bug still triggers. Can / should we get more testing for this?

**Comment by xaverdh** [comment]
closed in favour of #92081

**Pull request closed without merging**